index.js

```diff
@@ -65,6 +65,10 @@ const cpFile = (sourcePath, destinationPath, options) => {
 	const progressEmitter = new ProgressEmitter(path.resolve(sourcePath), path.resolve(destinationPath));
 	const promise = cpFileAsync(sourcePath, destinationPath, options, progressEmitter);
 
+	if (typeof options.onProgress === 'function') {
+		progressEmitter.on('progress', options.onProgress);
```

Comment:
It should be the inverse. `.on` should use `onProgress`. This would make it easier to remove the deprecated `.on` method in a future release.

Reply:
Sorry I don't understand. How can `promise.on` use `options.onProgress`?

Reply:
Pretend that `promise.on` doesn't exist and try to implement `options.onProgress`. That what the implementation should be.

Reply:
It makes sense but `promise.on` handles it on a different phase (chained). I can try to rewrite it this week.

Reply:
I've found a way to cover deprecated implementation through options object mutation but I am not sure about that.